proxy/http/HttpTransactHeaders.cc

```diff
@@ -683,82 +683,24 @@ HttpTransactHeaders::insert_server_header_in_response(const char *server_tag, in
   }
 }
 
-/// write the protocol stack to the @a via_string.
-size_t
-write_via_protocol_stack(char *via_string, size_t len, TSViaTransportVerbosity detail, ts::StringView *proto_buf, int nproto)
-{
-  char *via   = via_string; // keep original pointer for size computation later.
-  char *limit = via_string + len;
-  static constexpr ts::StringView tls_prefix{"tls/", ts::StringView::literal};
-
-  if (nproto <= 0 || via == nullptr || len <= 0) {
-    // nothing
-  } else if (detail == TS_VIA_TRANSPORT_FULL) {
-    ts::StringView *v = proto_buf;
-    for (int i = 0; i < nproto && (via + v->size() + 1) < limit; ++i, ++v) {
-      if (i)
-        *via++ = '-';
-      memcpy(via, v->ptr(), v->size());
-      via += v->size();
-    }
-    *via++ = ' ';
-  } else if (detail == TS_VIA_TRANSPORT_COMPACT) {
-    ts::StringView *proto_end = proto_buf + nproto;
-    bool http_1_0_p           = std::find(proto_buf, proto_end, IP_PROTO_TAG_HTTP_1_0) != proto_end;
-    bool http_1_1_p           = std::find(proto_buf, proto_end, IP_PROTO_TAG_HTTP_1_1) != proto_end;
-
-    if ((http_1_0_p || http_1_1_p) && via + 10 < limit) {
-      bool tls_p = std::find_if(proto_buf, proto_end, [](ts::StringView tag) { return tls_prefix.isPrefixOf(tag); }) != proto_end;
-      bool http_2_p = std::find(proto_buf, proto_end, IP_PROTO_TAG_HTTP_2_0) != proto_end;
-
-      memcpy(via, "http", 4);
-      via += 4;
-      if (tls_p)
-        *via++ = 's';
-      *via++   = '/';
-      if (http_2_p) {
-        *via++ = '2';
-      } else if (http_1_0_p) {
-        memcpy(via, "1.0", 3);
-        via += 3;
-      } else if (http_1_1_p) {
-        memcpy(via, "1.1", 3);
-        via += 3;
-      }
-      *via++ = ' ';
-    }
-  }
-  return via - via_string;
-}
-
 /// Look up the protocol stack and write it to the @a via_string.
 size_t
-write_client_protocol_stack(char *via_string, size_t len, HttpTransact::State *s)
+write_client_protocol_stack(HttpTransact::State *s, char *via_string, size_t len)
```

Comment:
This needs to be removed and reverted. I would grab the entire file from before #1534 or from #2018.